packages/flutter_tools/lib/src/asset.dart

```diff
@@ -332,74 +335,80 @@ class ManifestAssetBundle implements AssetBundle {
       primary: true,
     );
 
-    // Add fonts, assets, and licenses from packages.
+    // Add fonts, assets, and licenses from packages in the project's
+    // dependencies.
+    // To avoid bundling assets from dev_dependencies and other pub workspace
+    // packages, we compute the set of transitive dependencies.
+    final Map<String, Dependency> transitiveDependencies = computeTransitiveDependencies(
+      flutterProject,
+      packageConfig,
+      _fileSystem,
+      _logger,
+    );
     final Map<String, List<File>> additionalLicenseFiles = <String, List<File>>{};
-    for (final Package package in packageConfig.packages) {
-      final Uri packageUri = package.packageUriRoot;
-      if (packageUri.scheme == 'file') {
-        final String packageManifestPath = _fileSystem.path.fromUri(
-          packageUri.resolve('../pubspec.yaml'),
-        );
-        inputFiles.add(_fileSystem.file(packageManifestPath));
-        final FlutterManifest? packageFlutterManifest = FlutterManifest.createFromPath(
-          packageManifestPath,
-          logger: _logger,
-          fileSystem: _fileSystem,
+    for (final String packageName in transitiveDependencies.keys) {
+      final Package? package = packageConfig[packageName];
+      if (package == null) {
+        // This can happen with eg. `flutter run --no-pub`.
+        throwToolExit('Could not locate package:$packageName. Try running `flutter pub get`.');
+      }
```

Comment:
Can you expand on this comment? From my understand it is because `computeTransitiveDependencies` (as implemented) might return packages listed in `{CURRENT_APP}/pubspec.yaml` that are not resolvable to a package (i.e. because no package config is present or it is out of date).

(If that's not true, let's expand on why).